hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieParquetDataBlock.java

```diff
@@ -109,7 +112,7 @@ protected byte[] serializeRecords(List<HoodieRecord> records) throws IOException
 
     Schema writerSchema = new Schema.Parser().parse(super.getLogBlockHeader().get(HeaderMetadataType.SCHEMA));
     ByteArrayOutputStream baos = new ByteArrayOutputStream();
-    try (FSDataOutputStream outputStream = new FSDataOutputStream(baos, null)) {
+    try (SeekableDataOutputStream outputStream = storage.makeOutputSeekable(baos)) {
```

Comment:
Should we add a new `HoodieFileWriterFactory#getFileWriter` that takes `ByteArrayOutputStream`, instead of relying on `SeekableDataOutputStream`, to avoid adding `storage.makeOutputSeekable(baos)`?